aws-lc-rs/tests/data/ecdsa_sign_asn1_tests.txt

```diff
@@ -15,3 +15,39 @@ d = 519b423d715f8b581f4fa8ee59f4771a5b44c8130b4e3eacca54a56dda72b464
 Q = 041ccbe91c075fc7f4f033bfa248db8fccd3565de94bbfb12f3c59ff46c271bf83ce4014c68811f9a21a1fdb2c0e6113e06db7ca93b7404e78dc7ccd5ca89a4ca9
 k = 94a1bbb14b906a61a280f245f9e93c7f3b4a6247824f5d33b9670787642a68de
 Sig = 3046022100f3ac8061b514795b8843e3d6629527ed2afd6b1f6a555a7acabb5e6f79c8c2ac0221008bf77819ca05a6b2786c76262bf7371cef97b218e96f175a3ccdda2acc058903
+
+# [P-384,SHA3-384]
+
+Curve = P-384
+Digest = SHA3-384
+Msg = 6b45d88037392e1371d9fd1cd174e9c1838d11c3d6133dc17e65fa0c485dcca9f52d41b60161246039e42ec784d49400bffdb51459f5de654091301a09378f93464d52118b48d44b30d781eb1dbed09da11fb4c818dbd442d161aba4b9edc79f05e4b7e401651395b53bd8b5bd3f2aaa6a00877fa9b45cadb8e648550b4c6cbe
+d = 201b432d8df14324182d6261db3e4b3f46a8284482d52e370da41e6cbdf45ec2952f5db7ccbce3bc29449f4fb080ac97
+Q = 04c2b47944fb5de342d03285880177ca5f7d0f2fcad7678cce4229d6e1932fcac11bfc3c3e97d942a3c56bf34123013dbf37257906a8223866eda0743c519616a76a758ae58aee81c5fd35fbf3a855b7754a36d4a0672df95d6c44a81cf7620c2d
+k = dcedabf85978e090f733c6e16646fa34df9ded6e5ce28c6676a00f58a25283db8885e16ce5bf97f917c81e1f25c9c771
+Sig = 306502301c8b514baf0af1b6001d354a8ee0feae626375fe732ac586914e1cda52d49176057264c80e39b9860f8ce7302c0cf6aa023100cc720d7de789399b314787ac7786c6bc5e9c023be9e34fc54fb94b0257a796f5ef2bfc1d76a4f8e466d9c248cc075223
+
+Curve = P-384
+Digest = SHA3-384
+Msg = d768f41e6e8ec2125d6cf5786d1ba96668ac6566c5cdbbe407f7f2051f3ad6b1acdbfe13edf0d0a86fa110f405406b69085219b5a234ebdb93153241f785d45811b3540d1c37424cc7194424787a51b79679266484c787fb1ded6d1a26b9567d5ea68f04be416caf3be9bd2cafa208fe2a9e234d3ae557c65d3fe6da4cb48da4
+d = 23d9f4ea6d87b7d6163d64256e3449255db14786401a51daa7847161bf56d494325ad2ac8ba928394e01061d882c3528
+Q = 045d42d6301c54a438f65970bae2a098cbc567e98840006e356221966c86d82e8eca515bca850eaa3cd41f175f03a0cbfd4aef5a0ceece95d382bd70ab5ce1cb77408bae42b51a08816d5e5e1d3da8c18fcc95564a752730b0aabea983ccea4e2e
+k = 67ba379366049008593eac124f59ab017358892ee0c063d38f3758bb849fd25d867c3561563cac1532a323b228dc0890
+Sig = 306402303c7929216f5d4d7de1fd11ce6d230ae72d1b5c175d070af74b54f18b613fe4f7d943d26d97af4296f62340392a37103302306eba9f78ed70253e11cec580f704b8c86757a1c1b87a65a633c31edcad886e5111c5402b1ed1eac27a4ba9507bb5f555
```

Comment:
What is the source of these test vectors? Does the source for the vectors provide information about what these two vectors for P-384 & SHA3-384 are testing specifically?

Reply:
I could not find any official NIST/FIPS test vectors for P-384 & SHA3-384. 

These test vectors were produced by parsing the existing (P-384, SHA384) test vectors and generating new (P-384, SHA3-384) test vectors, each with an appropriate signature for the given parameters.

Reply:
>These test vectors were produced by parsing the existing (P-384, SHA384) test vectors and generating new (P-384, SHA3-384) test vectors, each with an appropriate signature for the given parameters.

how did you generate the new test vectors?

also, is the header of the file still correct?
```
# Tests from NIST CAVP 186-4 ECDSA2VS Test Vectors, Signature Generation Test
# http://csrc.nist.gov/groups/STM/cavp/documents/dss/186-3ecdsatestvectors.zip
#
#  CAVS 11.2
#  "SigVer" information for "ecdsa_values"
#  Curves/SHAs selected: P-224,SHA-224 P-224,SHA-256 P-224,SHA-384 P-224,SHA-512 P-256,SHA-224 P-256,SHA-256 P-256,SHA-384 P-256,SHA-512 P-384,SHA-224 P-384,SHA-256 P-384,SHA-384 P-384,SHA-512 P-521,SHA-224 P-521,SHA-256 P-521,SHA-384 P-521,SHA-512 K-233,SHA-224 K-233,SHA-256 K-233,SHA-384 K-233,SHA-512 K-283,SHA-224 K-283,SHA-256 K-283,SHA-384 K-283,SHA-512 K-409,SHA-224 K-409,SHA-256 K-409,SHA-384 K-409,SHA-512 K-571,SHA-224 K-571,SHA-256 K-571,SHA-384 K-571,SHA-512 B-233,SHA-224 B-233,SHA-256 B-233,SHA-384 B-233,SHA-512 B-283,SHA-224 B-283,SHA-256 B-283,SHA-384 B-283,SHA-512 B-409,SHA-224 B-409,SHA-256 B-409,SHA-384 B-409,SHA-512 BB-571,SHA-224 B-571,SHA-256 B-571,SHA-384 B-571,SHA-512
#  Generated on Tue Aug 16 15:27:42 2011
```

Reply:
> how did you generate the new test vectors?
I used a Ruby script (which interfaces with OpenSSL) to generate the new signatures. I will send you the script.

> also, is the header of the file still correct?
No, not any more at least. I'll remove this header. This file originated from [ring's test data](https://github.com/briansmith/ring/blob/95948b3977013aed16db92ae32e6b8384496a740/src/ec/suite_b/ecdsa/ecdsa_sign_asn1_tests.txt)

Reply:
ok, you can also just update the header to specify which tests come from where -- some from NIST and some generated by your script. It'd be good to also add the script source code to the file so others can reuse it in future if needed.

Reply:
I moved the newly created SHA3 test vectors into separate/dedicated data files in the latest commit.